controllers/subscription_controller.go

```diff
@@ -67,15 +68,29 @@ type SubscriptionReconciler struct {
 //+kubebuilder:rbac:groups=operators.coreos.com,resources=clusterserviceversions/finalizers,verbs=update
 //+kubebuilder:rbac:groups=operators.coreos.com,resources=installplans,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=operators.coreos.com,resources=operatorconditions,verbs=get;list;watch
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch
 //+kubebuilder:rbac:groups=admissionregistration.k8s.io,resources=mutatingwebhookconfigurations,verbs=get;list;watch;create;update
 
 func (r *SubscriptionReconciler) Reconcile(ctx context.Context, _ ctrl.Request) (ctrl.Result, error) {
-
 	logger := log.FromContext(ctx)
+	logger.Info("starting reconcile")
 
-	olmPkgRecords := getOlmPkgRecord()
+	configmap, err := GetOdfConfigMap(ctx, r.Client, logger)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
 
-	if err := r.setOperatorCondition(logger, olmPkgRecords); err != nil {
+	var olmPkgRecords []*OlmPkgRecord
+	var csvNames []string
+	ParseOdfConfigMapRecords(logger, configmap, func(record *OlmPkgRecord) {
+		if record.Pkg != "" && record.Csv != "" && record.Channel != "" {
```

Comment:
Why do we need this check? And if we do I would suggest to at least report something in the log saying that we are skipping a specific key

Reply:
If there is a mistake in the keys of the parsed data, we should skip that, as using it will have unexpected results. Added the loggging.